packages/react-core/src/Authenticator/hooks/useAuthenticator/__tests__/utils.spec.tsx

```diff
@@ -0,0 +1,60 @@
+import { AuthenticatorServiceFacade } from '@aws-amplify/ui';
+
+import { areSelectorDepsEqual, getComparator } from '../utils';
+
+describe('areSelectorDepsEqual', () => {
+  it('returns false when dep arrays have different lengths', () => {
+    const output = areSelectorDepsEqual([0, {}], [0]);
+
+    expect(output).toBe(false);
+  });
+
+  it('returns false when comparing an empty array and an empty object', () => {
+    const output = areSelectorDepsEqual([{}], [[]]);
+
+    expect(output).toBe(false);
+  });
+
+  it('returns false when dep arrays have similar object values', () => {
+    const output = areSelectorDepsEqual([0, { id: 4 }], [0, { id: 4 }]);
+
+    expect(output).toBe(false);
+  });
+
+  it('returns false when nested object values are different', () => {
+    const output = areSelectorDepsEqual(
+      [0, { options: {} }],
+      [0, { options: {} }]
```

Comment:
LGTM, I wonder we should eventually do deep equality checks. But since that's gonna run on every xstate change, I'm good with avoiding deep equality check computations.